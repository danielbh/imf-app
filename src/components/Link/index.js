import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const A = styled.a`
   text-decoration: none;
   display: block;
   margin: 2px;
`;

const SPAN = styled.span`
   text-decoration: underlined;
   display: block;
   margin: 2px;
`;

const Link = ({ active, children, onClick }) => {

  if (active) {
    return <SPAN className="btn btn-primary">{children}</SPAN>;
  }

  return (
    <A
      className="btn btn-default"
      href="#"
      onClick={e => {
        e.preventDefault();
        onClick();
      }}
    >
      {children}
    </A>
  );
};

Link.propTypes = {
  active: PropTypes.bool.isRequired,
  children: PropTypes.node.isRequired,
  onClick: PropTypes.func.isRequired,
};

export default Link;