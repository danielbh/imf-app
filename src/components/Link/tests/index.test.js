/**
 * Created by danielhollcraft on 9/8/17.
 */
import React from 'react';
import { shallow } from 'enzyme';
import Link from '../index'

const renderComponent = (props) => shallow(
  <Link {...props} />
);

describe('<Link />', () => {
  it('matches snapshot', () => {
    expect(renderComponent()).toMatchSnapshot();
  });

  it('displays span when active is true', () => {
    expect(renderComponent({active: true}).type()).toEqual('span');
  });

  it('displays anchor tag when active is false', () => {
    expect(renderComponent({active: false}).type()).toEqual('a');
  });

  it('calls onClick callback', () => {
    const onClick = jest.fn();
    renderComponent({onClick}).simulate('click', { preventDefault: () => {} });
    expect(onClick).toHaveBeenCalled();
  });

  it('calls preventDefault() on event object in onClick callback', () => {
    const preventDefault = jest.fn();
    renderComponent({onClick: () => {}}).simulate('click', { preventDefault });
    expect(preventDefault).toHaveBeenCalled();
  });
});