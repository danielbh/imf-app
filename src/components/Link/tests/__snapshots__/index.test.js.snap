// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> matches snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <a
    href="#"
    onClick={[Function]}
/>,
  "nodes": Array [
    <a
      href="#"
      onClick={[Function]}
/>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Link />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <a
          href="#"
          onClick={[Function]}
/>,
        "_debugID": 2,
        "_renderedOutput": <a
          href="#"
          onClick={[Function]}
/>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Link />,
}
`;
